subsys/fs/Kconfig.fatfs

```diff
@@ -14,6 +14,20 @@ if FAT_FILESYSTEM_ELM
 menu "ELM FAT file system settings"
 	visible if FAT_FILESYSTEM_ELM
 
+config FS_FATFS_MKFS
+	bool
+	help
+	  Enables mkfs support within ELM FAT FS.
+
+config FS_FATFS_MOUNT_MKFS
```

Comment:
It is on the FAT file system level because it is often used on external devices that you may not want to be formatted by your device when connected. It allows stripping a few kib, from ROM, of code you are never going to use...

Reply:
OK, the comment on size reduction has been missing...

Reply:
What external devices?  Is this for somebody plugging a USB thumb drive into a device running Zephyr?

If there's a good reason, ok, though I still think FS-wide would be more natural (what if Zephyr's configured for a littlefs file system? then the external device will still be formatted).  Also the impact of this on the mount flags does need to be made clear, probably both at the Kconfig and at the flags.

Reply:
> 
> 
> What external devices? Is this for somebody plugging a USB thumb drive into a device running Zephyr?
> 
Or SD.

>If there's a good reason, ok, though I still think FS-wide would be more natural (what if Zephyr's configured for a >littlefs file system?

Yes, but what if you have a device that internally uses LittleFS for configuration and calibration, etc. and will format internal partition when placed in burn-in environment. In such case you may need to have mkfs capability in LittleFS, while does not desire such within FAT FS or other systems.

Reply:
If you don't want mkfs for FatFs you disable `FS_FATFS_MKFS`.  If you have mkfs support in FatFs, and you don't want a particular mount to autoformat, you turn it off in the mount options, regardless of file system type.

I'm still not seeing why FatFs alone should have `FS_FATFS_MOUNT_MKFS` to disable auto-format completely when mkfs is available, regardless of mount flags that may request it.

Can you provide a more detailed use case for this?

Reply:
> 
> 
> If you don't want mkfs for FatFs you disable `FS_FATFS_MKFS`. If you have mkfs support in FatFs, and you don't want a particular mount to autoformat, you turn it off in the mount options, regardless of file system type.
> 
I can reduce code size bu rejecting code I will never use.

> I'm still not seeing why FatFs alone should have `FS_FATFS_MOUNT_MKFS` to disable auto-format completely when mkfs is available, regardless of mount flags that may request it.
> 
OK, so I have not predicted usage of LittleFS on external device, which is my oversight, and I find no other way to initialize  LittleFS on soc, and I need it.

> Can you provide a more detailed use case for this?
You are building device that displays simple led matrix animations from SD card and information like birth day dates, current time, some other calendar events, it stores data you want to be persistent, even if SD is removed, on internal littlefs formatted partition; it uses SD card in read-only mode for getting animations; formatting the SD is useless, e.g. when system is broken, as it will contain no images afterwards anyway.

Reply:
If the device never formats the SD card, why not just turn off FAT mkfs support?  If the device needs FAT mkfs support for another filesystem, why not just set the mount flags for the SD card to inhibit formatting that mount?  Why do you need to have a special flag that says "mkfs is supported, but don't allow it to be used in mounts even if the mount specifies it's ok"?

I must still be missing something....

Reply:
I think prompt there should be "**Allow** to format unformatted volume when mounting fails" or **Support**

`FS_FATFS_MKFS` is switch for enabling MKFS procedures in fatfs - Should be considered that this option might depend on availability of MKFS procedures(`depend on FS_FATFS_MKFS`), if volume formatting without  mounting FS immediately can e a user-case.

Reply:
OK, I get it now: the first one is the hidden one that forwards to FatFs; this one is the public one that would be set to `n` if formatting on mount is never desired.